clap_complete/src/engine/complete.rs

```diff
@@ -157,93 +157,7 @@ fn complete_arg(
-                completions.extend(hidden_longs_aliases(cmd));
-            } else if let Some((flag, value)) = arg.to_long() {
-                if let Ok(flag) = flag {
-                    if let Some(value) = value {
-                        if let Some(arg) = cmd.get_arguments().find(|a| a.get_long() == Some(flag))
-                        {
-                            completions.extend(
-                                complete_arg_value(value.to_str().ok_or(value), arg, current_dir)
-                                    .into_iter()
-                                    .map(|comp| comp.add_prefix(format!("--{flag}="))),
-                            );
-                        }
-                    } else {
-                        completions.extend(longs_and_visible_aliases(cmd).into_iter().filter(
-                            |comp| comp.get_value().starts_with(format!("--{flag}").as_str()),
-                        ));
-                        completions.extend(hidden_longs_aliases(cmd).into_iter().filter(|comp| {
-                            comp.get_value().starts_with(format!("--{flag}").as_str())
-                        }));
-                    }
-                }
-            } else if let Some(short) = arg.to_short() {
-                if !short.is_negative_number() {
-                    // Find the first takes_values option.
-                    let (leading_flags, takes_value_opt, mut short) = parse_shortflags(cmd, short);
-
-                    // Clone `short` to `peek_short` to peek whether the next flag is a `=`.
-                    if let Some(opt) = takes_value_opt {
-                        let mut peek_short = short.clone();
-                        let has_equal = if let Some(Ok('=')) = peek_short.next_flag() {
-                            short.next_flag();
-                            true
-                        } else {
-                            false
-                        };
-
-                        let value = short.next_value_os().unwrap_or(OsStr::new(""));
-                        completions.extend(
-                            complete_arg_value(value.to_str().ok_or(value), opt, current_dir)
-                                .into_iter()
-                                .map(|comp| {
-                                    let sep = if has_equal { "=" } else { "" };
-                                    comp.add_prefix(format!("-{leading_flags}{sep}"))
-                                }),
-                        );
-                    } else {
-                        completions.extend(
-                            shorts_and_visible_aliases(cmd)
-                                .into_iter()
-                                .map(|comp| comp.add_prefix(format!("-{leading_flags}"))),
-                        );
-                    }
-                }
-            }
+            complete_option(&mut completions, arg, cmd, current_dir);
```

Comment:
The other intermediate completion functions return a `Vec`. The performance of doing so here is unlikely to make a difference such that we should probably do so here